alembic/versions/2024_09_27_d6474325c722_refactor_pacbio_models.py

```diff
@@ -0,0 +1,103 @@
+"""Refactor pacbio models
+
+Revision ID: d6474325c722
+Revises: 18dbadd8c436
+Create Date: 2024-09-27 15:40:49.040271
+
+"""
+
+import sqlalchemy as sa
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "d6474325c722"
+down_revision = "18dbadd8c436"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # PacBioSequencingRun
+    op.add_column(
+        table_name="pacbio_sequencing_run",
+        column=sa.Column("barcoded_hifi_reads", sa.INTEGER, nullable=True),
+    )
+    op.add_column(
+        table_name="pacbio_sequencing_run",
+        column=sa.Column("barcoded_hifi_reads_percentage", sa.FLOAT, nullable=True),
+    )
+    op.add_column(
+        table_name="pacbio_sequencing_run",
+        column=sa.Column("barcoded_hifi_yield", sa.INTEGER, nullable=True),
+    )
+    op.add_column(
+        table_name="pacbio_sequencing_run",
+        column=sa.Column("barcoded_hifi_yield_percentage", sa.FLOAT, nullable=True),
+    )
+    op.add_column(
+        table_name="pacbio_sequencing_run",
+        column=sa.Column("barcoded_hifi_mean_read_length", sa.INTEGER, nullable=True),
+    )
+    op.add_column(
+        table_name="pacbio_sequencing_run",
+        column=sa.Column("unbarcoded_hifi_reads", sa.INTEGER, nullable=True),
+    )
+    op.add_column(
+        table_name="pacbio_sequencing_run",
+        column=sa.Column("unbarcoded_hifi_yield", sa.INTEGER, nullable=True),
+    )
+    op.add_column(
+        table_name="pacbio_sequencing_run",
+        column=sa.Column("unbarcoded_hifi_mean_read_length", sa.INTEGER, nullable=True),
+    )
+
+    # PacBioSampleSequencingMetrics
+    op.add_column(
+        table_name="pacbio_sample_run_metrics",
+        column=sa.Column("polymerase_mean_read_length", sa.INTEGER, nullable=True),
+    )
+    op.add_column(
+        table_name="pacbio_sample_run_metrics",
+        column=sa.Column("barcode", sa.VARCHAR(length=32), nullable=True),
+    )
```

Comment:
Should these all be nullable? I feel like they should not be. For non-barcoded runs we should be able to assign values to them still

Reply:
We discussed this a bit and given that there already are entries, we would need to give them some value. We should probably make them not nullable in the future though

Reply:
Exactly, I think the moment we do the alembic upgrade, it will fail if we make this nullable. We need to make this not nullable as soon as we ensure the existing entries have values on those columns. Do you agree? @Vince-janv

Reply:
Decided to keep nullable=True and fix later